### PR TITLE
DDF-4951 Changed methods that invoke security checks to final

### PR DESCRIPTION
**Pull request by tyler30clemens**
#### What does this PR do?
Non-final methods that perform security checks, or elevate code and then call other non-final methods, may be overridden in ways to bypass those security checks. This PR fixes multiple instances of that. 

#### Who is reviewing it? 
@austinsteffes 
@ahoffer 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@stustison

#### How should this be tested?
CI build and testers discretion 

#### What are the relevant tickets?
Fixes: #4951 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by tyler30clemens** [comment]
build now

File: platform/security/sts/security-sts-attributequerycommon/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (bakejeyner) [inline platform/security/sts/security-sts-attributequerycommon/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: ❓ What does this guy do?

**Comment by brjeter** [comment]
@bakejeyner theoretically there are other methods that should be marked final to better lockdown the codebase, but these ones in particular were found by static analysis.

File: platform/security/sts/security-sts-attributequerycommon/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (tyler30clemens) [reply platform/security/sts/security-sts-attributequerycommon/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: In general, mockito does not allow final methods to be mocked and stubbed. This (along with the dependency change) upgrades mockito to the version that allows for this to occur.

**Comment by jdelta-RBS** [comment]
As far as versions affected by this, would that be 2.15.0 only? Or would it include e.g. 2.13.10?

**Pull request merged**